### PR TITLE
Adopt EditorContext parameter. Switch to boolean. Handle some TODOs

### LINKED ISSUE #391: Adopt `Utils#canBeHeadOfStream(graph,element)` method for TAPing
`Utils#canBeHeadOfStream(graph,element)` works based `source` apps and apps having input link from a `tap` port.

- `source` element can still be a head of stream, but since `tap` port it should probably be a `tap` link (link.attr('props/tap`) === 'true'`? Should be just `true/false` boolean)? 
- Any other conditions? 
- Channel being the head of stream?

**Tasks:**
- [ ] Change data type from String to Boolean
- [ ] Revisit how we create nodes/links (we should ask the API to do it for us)

### PR DESCRIPTION
**Pull request by BoykoAlex**
Fixes #391

### CODE REVIEW

File: ui/src/app/streams/flo/render.service.ts [hunk 1]
```diff
@@ -256,9 +256,9 @@ export class RenderService implements Flo.Renderer {
 
     initializeNewLink(link: dia.Link, viewerDescriptor: Flo.ViewerDescriptor) {
         link.set('smooth', true);
-        link.attr('metadata/metadata/unselectable', 'true');
+        link.attr('metadata/metadata/unselectable', true);
         // var isTapLink = link.attr('props/isTapLink');
-        // if (isTapLink === 'true') {
+        // if (isTapLink) {
```
Comment (jvalkeal) [inline ui/src/app/streams/flo/render.service.ts]: Remove dead code around this.

File: ui/src/app/streams/flo/render.service.ts [hunk 2]
```diff
@@ -481,115 +488,93 @@ export class RenderService implements Flo.Renderer {
...
         }
-        this.refreshVisuals(link, 'props/isTapLink', paper);
+        this.refreshVisuals(link, 'props/isTapLink', flo.getPaper());
         // if (source) {
         //     var outputLinks = graph.getConnectedLinks(source, {outbound: true});
         //     var isPrimaryLink = true;
         //     for (var i=0;i<outputLinks.length;i++) {
         //         var ol = outputLinks[i];
-        //         if (ol === link || (ol.attr('props/isTapLink')==='true')) {
+        //         if (ol === link || (ol.attr('props/isTapLink'))) {
```
Comment (jvalkeal) [inline ui/src/app/streams/flo/render.service.ts]: Remove dead code around this.

**Comment by BoykoAlex** [comment]
@jvalkeal looks like you wanted the commented out code to be gone - removed it.

**Comment by jvalkeal** [comment]
Merged per https://github.com/spring-cloud/spring-cloud-dataflow-ui/commit/26cf80a39294a7e598c2257fc1603dbbef8bc234

**Pull request closed without merging**